### PR TITLE
Flaggers

### LINKED ISSUE #5: Design a interface for analyzers
In order for this story to be complete the following criteria should be completed,

-A document describing the interface and how to create a new analyzer should be created
-A basic analyzer that  does sanity checks is created
-The interface allows for the following
  -A line of data from the portal database should be accepted as an input
  -The interface should return a list of flags that the data triggered or have a simple way to retrieve it

### PR DESCRIPTION
**Pull request by tinlun**
Closes #5, and whatever # @gmillia was working on.

These commits are combined efforts from me and @gmillia .

### CODE REVIEW

**Comment by sawyerwatts** [comment]
The. gitignore looks generic (not an insult), did you pull it from somewhere?

File: diff [hunk 1]
```diff
@@ -0,0 +1,9 @@
+diff --git a/.gitignore b/.gitignore
+index 762cc89..9f724da 100644
+--- a/.gitignore
++++ b/.gitignore
+@@ -133,3 +133,4 @@ dmypy.json
+ 
+ # pytype static type analyzer
+ .pytype/
++*.swp
```
Comment (sawyerwatts) [inline diff]: What does this file do? Trying to google it just returns information about git-diff.

Comment (tinlun) [reply diff]: yes, .gitignore pulled from https://github.com/github/gitignore/blob/master/Python.gitignore
*.swp is vim swap files.

Comment (sawyerwatts) [reply diff]: I don't mean swap files specifically, I mean the file "diff" in general. I didn't want to need to select the whole file.

Comment (tinlun) [reply diff]: oh, yeah i don't think that's supposed to be there.

File: flags.txt [hunk 2]
```diff
@@ -0,0 +1,35 @@
+There are different types of flags used to represent different types of things present in a row data (object):
+
```
Comment (sawyerwatts) [inline flags.txt]: I think this makes sense in docs/; what are your thoughts? I know we're focusing more on getting code up right now, as opposed to putting it into src/ and the like, but if it's just a single text file, could we just do it now?

Comment (tinlun) [reply flags.txt]: I agree.

File: flaggers/null.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+#data is a row of data from the db: parsed JSON
+class Null(Flagger):
+  name = 'Null'
+  columns_flag_dict = {
+    'service_date' : Flags.SERVICE_DATE_NULL,
+    'vehicle_number' : Flags.VEHICLE_NUMBER_NULL,
+    'leave_time' : Flags.LEAVE_TIME_NULL,
+    'train' : Flags.TRAIN_NULL,
+    'badge' : Flags.BADGE_NULL,
```
Comment (sawyerwatts) [inline flaggers/null.py]: The same comment about badge and row_id applies here as well.

**Pull request merged**